src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/CSharpBlockTest.cs

```diff
@@ -283,7 +283,7 @@ public void CorrectlyParsesDoWhileBlockMissingWhileConditionWithSemicolon()
         [Fact]
         public void CorrectlyParsesDoWhileBlockMissingWhileClauseEntirely()
         {
-            ParseBlockTest("do { var foo = bar; } narf;");
+            ParseBlockTest("do { var foo = bar; } narf;", expectedParseLength: 21);
```

Comment:
I've only modified a sample set of the failing tests just to give an idea. I can change all the tests if there is no objection for the way this is tested.

Reply:
There are 70 more failing tests to fix.

Reply:
Yeah this is roughly what I had in mind 👍